website/docs/pubsub.md

```diff
@@ -0,0 +1,235 @@
+
+Limitations of getting audit violations from audit logs:
+
+- It could be difficult to parse audit pod logs to look for violation messages, as violation logs would be mixed together with other log statements. Additionally, when there are huge number of violations, it’s possible to miss part of the log as the pod logs get rotated.
+
+This feature uses publish and subscribe (pubsub) model that allows Gatekeeper to export audit violations over a broker that can be consumed by a subscriber independently. Therefore, it allows users to get all the audit violations.
+
+## Enabling Gatekeeper to export audit violations
+
+Install prerequisites such as a pubsub tool, a message broker etc.
+
+### Setting up audit with pubsub enabled
+
+In the audit deployment, set the `--enable-pub-sub` flag to `true` to publish audit violations. Additionally, `--audit-connection` and `--audit-channel` flags must be set to allow audit to publish violations. `--audit-connection` must be set to the name of the connection config, and `--audit-channel` must be set to name of the channel where violations should get published.
+
+Create a connection configMap that supplies appropriate set of configurations for a connection to get established. For instance, to establish a connection that uses Dapr to publish messages this configMap is appropriate:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: audit
+  namespace: gatekeeper-system
+data:
+  provider: "dapr"
+  config: |
+    {
+      "component": "pubsub"
+    }
+```
+
+- `provider` field in `configMap.data` determined which tool/driver should be used to establish a connection. Valid values are: `dapr`
+- `config` field in `configMap.data` is a json data that allows users to pass appropriate information to establish connection to use respective provider.
+
+#### Available Pubsub drivers
+Dapr: https://dapr.io/
+
+### Violations
+
+The audit pod publishes violations in following format:
+
+```json
+{
+  "id": "2023-07-18T21:21:52Z",
+  "details": {
+    "missing_labels": [
+      "test"
+    ]
+  },
+  "eventType": "violation_audited",
+  "group": "constraints.gatekeeper.sh",
+  "version": "v1beta1",
+  "kind": "K8sRequiredLabels",
+  "name": "pod-must-have-test",
+  "message": "you must provide labels: {\"test\"}",
+  "enforcementAction": "deny",
+  "resourceAPIVersion": "v1",
+  "resourceKind": "Pod",
+  "resourceNamespace": "nginx",
+  "resourceName": "nginx-deployment-cd55c47f5-2b84x",
+  "resourceLabels": {
+    "app": "nginx",
+    "pod-template-hash": "cd55c47f5"
+  }
+}
+```
+
+### Quick start with publishing violations using Dapr and Redis
```

Comment:
Do we want to take responsibility for making sure that this documentation remains current? What if Dapr and/or Redis change their installation methods? Should we just point users to Dapr's docs if they need a reference installation?

TBH if we wan't to have a "quick start" guide, I'd use something other than Dapr, since there would be less to install, and therefore less things to break. Maybe use Redis directly, as an example (though I still wouldn't want to own the accuracy of installation instructions for a 3rd party product).

Reply:
I am fine with pointing to just dapr docs. 

@sozercan @ritazh thoughts?